internal/server/iigo.go

```diff
@@ -89,7 +90,10 @@ func (s *SOMASServer) runIIGOAllocations() error {
 		allocation := v.RequestAllocation()
 
 		if allocation <= s.gameState.CommonPool {
-			s.giveResources(clientID, allocation, "allocation")
+			err := s.giveResources(clientID, allocation, "allocation")
+			if err != nil {
+				return errors.Errorf("Failed to give resources: %v", err)
```

Comment:
Just to confirm, will the program exit if it get's to here?

Reply:
yes. do you want it to not when it is trying to give NaN/negative?

Reply:
Oh right sorry, I thought it game error even when there isn't enough in common pool

Reply:
If it only produces errors on NaN/negative that's fine with me